python/cudf/cudf/core/indexing.py

```diff
@@ -266,11 +266,22 @@ def _getitem_tuple_arg(self, arg):
             else:
                 return columns_df.index._get_row_major(columns_df, arg[0])
         else:
-            df = DataFrame()
-            for col in columns_df.columns:
-                # need Series() in case a scalar is returned
-                df[col] = Series(columns_df[col].loc[arg[0]])
-            df.columns = columns_df.columns
+            if hasattr(arg[0], "__len__"):
+                arg = (column.as_column(arg[0]), arg[1])
```

Comment:
Is the goal to handle everything but the scalar case here? Doing `hasattr` checks are expensive so if possible it would be better to check if we have a scalar, otherwise call `as_column`.

Reply:
This is meant for slice/scalar and other non-column requirements.

Reply:
Got it. I think it may make more sense to handle scalar / slice specifically and then defer down to `as_column` otherwise.